### PR TITLE
175 thumbnail

### PR DESCRIPTION
**Pull request by connectbo**
This pr
- Refactored galleryview container and extract each item as component
- Detect the type of each item from rule engine and render icon accordingly, whether it is collection or static file. Once we know how we generate and visit thumbnails, I can update its path so real thumbnails can be displayed
- Clicking collection item will navigate user to that directory
- Add navigation on the top path directory bar, just like the collection page
- Add error handling when there is no irods rule engine or error occurs when fetching data

### CODE REVIEW

**Comment by connectbo** [comment]
Screen Shot 2021-05-04 at 4 03 50 PM

File: src/metalnx-web/src/main/javascript/gallery/components/Gallery.vue [hunk 1]
```diff
@@ -1,8 +1,20 @@
 <template>
   <!-- Search -->
-  <div>
+  <div v-if="!thumbnails">
+    Error occurs when fetching data from iRODS Rule Engine.
```
Comment (trel) [inline src/metalnx-web/src/main/javascript/gallery/components/Gallery.vue:4]: 'An error has occurred while fetching thumbnail data from iRODS.'

File: src/metalnx-web/src/main/javascript/gallery/components/Gallery.vue [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+  </div>
+  <div v-else-if="thumbnails.error">iRODS Rule Engine does not exist.</div>
```
Comment (trel) [inline src/metalnx-web/src/main/javascript/gallery/components/Gallery.vue:6]: 'An error has occurred while fetching thumbnail data from iRODS.'

?   can we be more specific... all we know is that it didn't come back good.  right?

**Review by trel — COMMENTED** [review]
can we still give them some navigation or something...

Screen Shot 2021-05-04 at 4 41 00 PM

 or bounce back to the regular listing if the thumbnails aren't configured correctly with this message in a little color error bar?

**Comment by trel** [comment]
If there is more than one replica and the replicas have different file sizes (regardless of their status, & or X...) then there is one icon per object/size combo in the icon view.

This is not the case for the regular browser list view - so we should consider how we want to display this scenario.

Goals:
 - only show one icon per logical data object
 - show size of 'good' replica if there are stales
 - show size of first stale replica if there are no good replicas

@alanking might need your replica brain a little here

question is how can we get this information in the query itself?  and/or how can bo filter his existing results to meet the goals listed above...

**Comment by connectbo** [comment]
I refactored several components and enhance the error handling we have.
- Polish path displaying (edge case: a long path)
- Polish gallery item naming (edge case: use 'ellipsis' to handle a really long name)
- Style error notifications inline with the collection browser. User will have the link to go back to collection browser. Attached two images if those would be helpful.

Screen Shot 2021-05-06 at 9 25 42 PM


Screen Shot 2021-05-06 at 9 03 25 PM

**Comment by trel** [comment]
i like it - will render it here locally later today.

i see a branch conflict - need to rebase it cleanly.

**Comment by connectbo** [comment]
> i like it - will render it here locally later today.
> 
> i see a branch conflict - need to rebase it cleanly.

You are right, I didn't notice that when it's wip. Will fix it.

**Comment by trel** [comment]
I think we are about ready for a squash with issue numbers... and we can get this merged.   Don't # them yet.

**Pull request merged**